iceoryx_posh/test/moduletests/test_roudi.cpp

```diff
@@ -0,0 +1,545 @@
+// Copyright (c) 2021 by Robert Bosch GmbH, Apex.AI Inc. All rights reserved.
```

Comment:
Concerning this file, many EXPECT are missing. I am aware but I didn't have any idea how to really check the different tests. On my side, I think it is missing an errorHandler when number of elements is incorrect.

Reply:
To which number of elements are you referring to?

Reply:
Sorry, I was not so explicit. In the procesManager method of class Roudi (roudi.cpp), if the message does not contain the correct number of elements depending on the MqMessageType received, an error is set. In my opinion, errohandler could be used there.

Reply:
ErrorHandler is not used right now. But, I guess it could be a potential improvement of the existing code, correct ?

Reply:
I guess you refer to the method `processMessage` right?
Yes the errorhandler should be used there, but only with level `Moderate` because RouDi can proceed and the request from the runtime is not processed.
There is even in one case the errorhandler used: https://github.com/eclipse-iceoryx/iceoryx/blob/f7fee3c237bfe8ae15da434aa78937dba0b0c8ec/iceoryx_posh/source/roudi/roudi.cpp#L230

@elBoberido does it make sense to have an error like `Error::kROUDI__WRONG_NUMBER_PARAMETERS_MQMESSAGE`? Or do you want to refactor that too?

Reply:
I think that's not necessary on the roudi side. It just hast to be communicated back to the application since this is input from an application and therefore the error handler should also be called on the application side.

Reply:
Does it mean the method process manager should return something ?

Reply:
This is something that will heavily be refactored with #332 in the future to give the application more information what's going on. As a side effect, this makes testing also easier.
Currently there is no response to the application if the number of parameter don't match. With #332 that will change and an error response will be sent back.
What's even worse, calling the error handler with `Error::kPORT_MANAGER__INTROSPECTION_MEMORY_MANAGER_UNAVAILABLE` is definitely wrong and should be removed. The application just provided us with the wrong number of parameter.
Currently the `RouDi` class is not really good testable.